compiler/src/iree/compiler/Dialect/Flow/Transforms/SetEncoding.cpp

```diff
@@ -233,10 +219,6 @@ class setContractionOpEncoding
     if (!linalgOp.hasPureTensorSemantics()) {
       return failure();
     }
-    if (getCompilationInfo(linalgOp)) {
-      return rewriter.notifyMatchFailure(
-          linalgOp, "the op has preset compilation strategy, skip SetEncoding");
-    }
```

Comment:
@hanhanW I have removed this check because I don't see the reason it is necessary and I don't want to add a dependency on Codegen in Flow. Is it okay to remove now?

Reply:
This was added in https://github.com/iree-org/iree/pull/15455

Reply:
Interesting... I thought that it would fail in the test: https://github.com/iree-org/iree/blob/main/tests/e2e/regression/lowering_config.mlir

Reply:
It's okay to remove it if it does not trigger the failure.

Reply:
Oh, it might fail. I didn't build test-deps locally

Reply:
The intention is that people might want to have customized codegen strategy for some ops, so they could put attributes on the operation. In the past, compilation_info was the only way for doing it. Today we have a new mechanism using transform dialect, which might improve the situation.

It's still a little valuable to me until we have a better setup. Ideally, we should wrap them with other ops. Then we will be able to drop the special check.

Reply:
Ok, I will leave it in for now and add the dependency. I'll leave a TODO comment to eventually remove it.

Reply:
> Maybe not, though. We don't SetEncoding on stablehlo.dot, so I don't think it should be an issue.

IIRC, they are lowered to linalg ops with those attributes, so we will likely hit the case.